torchtune/modules/attention.py

```diff
@@ -190,20 +191,18 @@ def forward(
         v = v.reshape(bsz, seq_len, -1, self.head_dim)
 
         # Apply positional embeddings
-        q = self.pos_embeddings(q, curr_pos)
-        k = self.pos_embeddings(k, curr_pos)
-
-        # Update key-value cache
-        if self.kv_cache is not None:
-            k, v = self.kv_cache.update(
-                bsz=bsz, seq_len=seq_len, curr_pos=curr_pos, k_val=k, v_val=v
-            )
+        q = self.pos_embeddings(q, input_pos)
+        k = self.pos_embeddings(k, input_pos)
 
         # [b, n_h, s, h_d]
         q = q.transpose(1, 2)
         k = k.transpose(1, 2)
         v = v.transpose(1, 2)
 
+        # Update key-value cache
+        if self.kv_cache is not None:
```

Comment:
Just for my understanding, any benefit in updating kv-cache post transpose? Does it enable better access patterns or the like?

Reply:
It's a good question - let me think a bit about it